crates/astria-composer/src/executor/mod.rs

```diff
@@ -415,6 +423,41 @@ impl Executor {
 
         reason.map(|_| ())
     }
+
+    // check for mismatched configured chain_id and sequencer client chain_id
+    pub(crate) async fn ensure_configured_chain_id_matches_remote(&self) -> eyre::Result<()> {
+        let retry_config = tryhard::RetryFutureConfig::new(u32::MAX)
+            .exponential_backoff(Duration::from_millis(100))
+            .max_delay(Duration::from_secs(20))
+            .on_retry(
+                |attempt: u32,
+                 next_delay: Option<Duration>,
+                 error: &sequencer_client::tendermint_rpc::Error| {
+                    let wait_duration = next_delay
+                        .map(humantime::format_duration)
+                        .map(tracing::field::display);
+                    warn!(
+                        attempt,
+                        wait_duration,
+                        error = error as &dyn std::error::Error,
+                        "attempt to fetch sequencer genesis info; retrying after backoff",
+                    );
+                    futures::future::ready(())
+                },
+            );
+        let client_genesis: tendermint::Genesis =
+            tryhard::retry_fn(|| self.sequencer_client.genesis())
+                .with_config(retry_config)
+                .await
+                .wrap_err("failed to retrieve sequencer genesis after many attempts")?;
+        ensure!(
+            self.sequencer_chain_id == client_genesis.chain_id.as_str(),
+            "mismatch in configured chain_id: {0} and sequencer chain_id: {1}",
```

Comment:
nit: this is accurate but concerned the message might not give enough context information to operator to understand how configuration is wrong.

Something like: 
`mismatch in chain id: config specifies {0}, but sequencer rpc is for {1}`.

Reply:
Done in be387af.